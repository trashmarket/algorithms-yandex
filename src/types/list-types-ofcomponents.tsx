import { ElementStates } from "./element-states";

export interface LetterStep {
    letter: string;
    state?: ElementStates;
    index: number
}

export interface TFibStep {
    index: number,
    fibNum: number
}