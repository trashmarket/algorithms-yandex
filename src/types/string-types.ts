import { ElementStates } from "./element-states";

export interface LetterStep {
    letter: string;
    state?: ElementStates;
    index: number
}