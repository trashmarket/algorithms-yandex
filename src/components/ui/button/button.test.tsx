import { render, fireEvent } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { Button, ButtonProps } from "./button";
import { Direction } from "../../../types/direction";
describe("button test ui", () => {

  it("button with text", () => {
    const { getByRole } = render(<Button text="Hello" />);
    const button = getByRole("button");
    expect(button).toMatchSnapshot();
  });

  it('button without text', () => {
    const { getByRole } = render(<Button/>);
    const butto = getByRole('button');
    expect(butto).toMatchSnapshot();
  })

  it('button disabled', () => {
    const { getByRole } = render(<Button disabled={true}/>);
    const butto = getByRole('button');
    expect(butto).toMatchSnapshot();
  })

  it('button is loder', () => {
    const { getByRole } = render(<Button isLoader={true}/>);
    const butto = getByRole('button');
    expect(butto).toMatchSnapshot();
  })

  it('check click', async () => {
    const mockHandleClick = jest.fn();
    const { getByRole,  } = render(<Button onClick={mockHandleClick}/>);
    const button = getByRole('button');

    expect(mockHandleClick).not.toBeCalled();

    await userEvent.click(button);

    expect(mockHandleClick).toBeCalled();

  })

  it('check sorting button', () => {
    const { getByRole } = render(<Button sorting={Direction.Ascending}/>);
    const button = getByRole('button');

    expect(button).toMatchSnapshot();
  })

  it('check linkedList small button', () => {
    const { getByRole } = render(<Button linkedList="small"/>);
    const button = getByRole('button');

    expect(button).toMatchSnapshot();
  })

  it('check linkedList big button', () => {
    const { getByRole } = render(<Button linkedList="big"/>);
    const button = getByRole('button');

    expect(button).toMatchSnapshot();
  })
});
