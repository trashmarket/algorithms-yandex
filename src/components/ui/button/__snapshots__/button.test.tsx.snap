// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button test ui button whith text 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    class="text undefined"
  >
    Hello
  </p>
</button>
`;

exports[`button test ui button with text 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    class="text undefined"
  >
    Hello
  </p>
</button>
`;
